packager/react-native-xcode.sh

```diff
@@ -10,11 +10,35 @@
 # This script is supposed to be invoked as part of Xcode build process
 # and relies on environment variables (including PWD) set by Xcode
 
+PLISTBUDDY='/usr/libexec/PlistBuddy'
+PLIST=$TARGET_BUILD_DIR/$INFOPLIST_PATH
+
+# 3 options for adding 'localhost' exception for ATS when using the iphonesimulator platform: add it, give a warning, or give an error and fail to build.
+ATS_LOCALHOST_EXCEPTION_MISSING="warn" # valid options: "add", "warn", "error"
+
 case "$CONFIGURATION" in
   Debug)
     # Speed up build times by skipping the creation of the offline package for debug
     # builds on the simulator since the packager is supposed to be running anyways.
     if [[ "$PLATFORM_NAME" = "iphonesimulator" ]]; then
+      localhost_exception=$($PLISTBUDDY -c "Print NSAppTransportSecurity:NSExceptionDomains:localhost:NSTemporaryExceptionAllowsInsecureHTTPLoads" "$PLIST")
+      if [[ "$localhost_exception" = "true" ]]; then
+        echo "Confirmed 'localhost' exception for App Transport Security in $PLIST"
+      else
+        SOURCE_PLIST="$SOURCE_ROOT/$PROJECT_NAME/Info.plist"
+        ATS_EXCEPTION_MSG="The development server will not be accessible without adding a 'localhost' exception for App Transport Security to the Info.plist file. To do this, perform the command '$PLISTBUDDY -c \"Add NSAppTransportSecurity:NSExceptionDomains:localhost:NSTemporaryExceptionAllowsInsecureHTTPLoads bool true\" \"$SOURCE_PLIST\"'"
+
+        if [[ "$ATS_LOCALHOST_EXCEPTION_MISSING" = "add" ]]; then
```

Comment:
I'd prefer a convention over configuration. Once we add the configuration option, we'd have to support it for a long long time